src/transformations/transform-constructor.ts

```diff
@@ -80,32 +91,43 @@ export function* transformConstructor(
       argsList = '',
       unchainedArgsList = '',
       argNames: string[] = [],
+      modifiers = '',
     ) => [
       `function __${name}_init(${argsList}) internal onlyInitializing {`,
-      buildSuperCallsForChain2(contractNode, tools, helper),
+      buildSuperCallsForChain(contractNode, tools, helper),
       emptyConstructor ? [] : [`__${name}_init_unchained(${argNames.join(', ')});`],
       `}`,
       ``,
-      `function __${name}_init_unchained(${unchainedArgsList}) internal onlyInitializing {`,
+      `function __${name}_init_unchained(${unchainedArgsList}) internal onlyInitializing ${modifiers}{`,
```

Comment:
Since we will be converting the modifier array into strings here this line will become very long. My recommendation is to do something like this:


```suggestion
      [
        'function',
        `__${name}_init_unchained(${unchainedArgsList})`,
        'internal onlyInitializing',
        ...modifiers.map(m => ...),
         '{',
       ].join(' '),
```